pubsub/gcppubsub/gcppubsub.go

```diff
@@ -122,6 +123,15 @@ func (o *lazyCredsOpener) defaultConn(ctx context.Context) (*URLOpener, error) {
 			o.err = err
 			return
 		}
+
+		// Connect to the GCP pubsub emulator by overriding the default endpoint
+		// if the 'PUBSUB_EMULATOR_HOST' environment variable is set.
+		// Check https://cloud.google.com/pubsub/docs/emulator for more info.
+		emulatorEndPoint := os.Getenv("PUBSUB_EMULATOR_HOST")
```

Comment:
Shouldn't this be inside `Dial`? Not all users use a URL opener.

Reply:
I think non URL opener user should provide the grpc connection to connect to the emulator host.
And when URL opener is used with the environment variable they will default to emulator host.

Reply:
I see. Yes, that seems OK.